### PR TITLE
fix(#344): plug Sentry capture leaks in error boundary + surface full error report

### LINKED ISSUE #344: Sentry rate des erreurs qui trigger l'ErrorBoundary (race init + chunk-load)
## Symptôme

Sur mobile, j'ai eu un écran ErrorBoundary sur `/historique`. Retry a corrigé. Pas de détail visible côté UI (gating `isDev`), donc impossible de noter l'erreur. **Et surtout : 0 event correspondant dans le dashboard Sentry**, alors que la DSN est bien branchée et que d'autres erreurs y arrivent normalement.

## Diagnostic

Le pipeline `AppErrorBoundary` → Sentry a au moins **3 trous** identifiés :

### 1. Race condition `initSentry` vs erreur (le plus probable)

Dans `src/main.tsx` :

```78:86:src/main.tsx
    runWhenIdle(() => {
      void import("@/lib/sentry")
        .then(({ initSentry }) => initSentry())
        .catch(() => {})
      ...
    })
```

`initSentry()` est planifié dans `requestIdleCallback` avec un timeout de 2s. Sur mobile (CPU saturé, network lent), idle peut traîner. Tant que `Sentry.init()` n'a pas tourné, **`captureException` est un no-op** (pas de Hub configuré) → l'event est dropped silencieusement.

Le commentaire en haut du fichier l'admet déjà :
> early-boot errors that fire before idle won't be captured — acceptable trade for LCP/TBT

Sauf que ce n'est pas que "early-boot" : c'est **toute la fenêtre tant que idle n'a pas tiré**. Sur un cold-load + navigation rapide vers `/historique`, on est typiquement dans ce trou.

### 2. Chunk-load failure (PWA stale cache)

Dans `src/components/AppErrorBoundary.tsx` :

```30:42:src/components/AppErrorBoundary.tsx
  componentDidCatch(error: Error, info: ErrorInfo): void {
    void import("@sentry/react")
      .then(({ captureException }) => {
        captureException(error, {
          contexts: {
            react: { componentStack: info.componentStack ?? undefined },
          },
        })
      })
      .catch(() => {
        // Sentry unavailable (network/offline) — swallow to avoid loops.
      })
  }
```

Si le SW sert une version cachée et que `import("@sentry/react")` pointe vers un hash qui n'existe plus côté serveur → **ChunkLoadError** → `.catch(() => {})` avale tout. Idem si l'erreur d'origine était elle-même un ChunkLoadError de navigation : le réseau qui pète te bouffe AUSSI le fetch du SDK Sentry.

### 3. UX : aucun fallback pour reporter manuellement

`ErrorFallback` gate le détail de l'erreur derrière `import.meta.env.DEV`. En prod l'utilisateur voit juste un message générique, sans Event ID, sans stack, sans rien à copier. Si Sentry rate, il n'y a **aucune info exploitable**.

### (a) Garantir que Sentry est init avant la capture — `componentDidCatch`

Idempotent (gardé par le DSN check), donc safe à appeler depuis `componentDidCatch` :

\`\`\`ts
componentDidCatch(error, info) {
  void Promise.all([
    import("@sentry/react"),
    import("@/lib/sentry"),
  ]).then(([{ captureException }, { initSentry }]) => {
    initSentry() // no-op si déjà init, sinon init synchrone après l'import
    captureException(error, { ... })
  }).catch(() => { /* still swallow */ })
}
\`\`\`

Ferme le trou #1. La pénalité ne se paye qu'au moment où il y a déjà une erreur — donc on s'en fout du TBT.

### (c) Surface complète de l'erreur en prod (filet de sécurité)

Dans \`ErrorFallback\` quand variant=\"page\" :

- **Event ID** copiable (hash court genre \`err_a7b3f2\` dérivé de \`message + timestamp\`)
- **error.message** brut visible (pas juste en DEV)
- Stack trace derrière toggle \"Détails techniques\" (aussi en prod)
- **Bouton \"Copier\"** qui copie un payload complet : \`{ id, message, stack, componentStack, route, userAgent, timestamp, appVersion }\`

Comme ça même si Sentry rate, on a un screenshot ou un paste exploitable. L'Event ID sert aussi de référence partagée entre user/dashboard si l'event a quand même fini par arriver.

## Hors scope (peut-être un follow-up)

(b) Fallback \`navigator.sendBeacon\` direct sur l'endpoint Sentry HTTP en cas de chunk-load fail — ~50 lignes de parsing DSN + tests, on attend de voir si (a) suffit avant.

## Acceptance criteria

- [ ] Erreur lancée avant le tir de \`requestIdleCallback\` est bien capturée par Sentry
- [ ] \`ErrorFallback\` page-variant affiche en prod : Event ID, message, stack toggle, bouton Copier
- [ ] Le payload copié contient route + userAgent + timestamp + appVersion
- [ ] Tests pour (a) et (c)
- [ ] Aucune régression sur les bytes shippés en happy path (Sentry reste hors du main bundle)

### PR DESCRIPTION
**Pull request by PierreTsia**
## What

- **Fix race in `AppErrorBoundary.componentDidCatch`**: now imports `@/lib/sentry` alongside `@sentry/react` and calls `initSentry()` before `captureException`. The Sentry event is tagged with `error_id` so the dashboard event lines up with the user-visible ID.
- **New `src/lib/errorReport.ts`** util: stable `makeErrorId` (`err_xxxxxx` derived from `error.message`), `buildErrorReport`, `formatReportAsMarkdown`.
- **`ErrorFallback` redesign**: error ID badge, raw `error.message`, technical-details toggle (stack + component stack) and a "Copy report" button — **all visible in production**, no longer gated by `import.meta.env.DEV`. Markdown payload bundles id, timestamp, route, app version, user agent, message, stack and component stack, ready to paste into a GH issue.

## Why

Reported on mobile: ErrorBoundary screen on `/historique`, retry fixed it — but **0 events** in the Sentry dashboard. The DSN is wired and other events flow normally, so Sentry was silently dropping this specific capture. Diagnosed three holes in the pipeline:

1. **Race condition `initSentry` vs error**. `initSentry()` is scheduled in `requestIdleCallback` (timeout 2s) in `main.tsx`. On mobile (busy CPU, slow network), idle can lag. While `Sentry.init()` hasn't run, `captureException` is a no-op against the global hub — event silently dropped. The existing comment ("early-boot errors that fire before idle won't be captured") underestimated the size of that window.
2. **Chunk-load failures** (PWA stale cache, offline). `import("@sentry/react").catch(() => {})` swallows everything; if the original error was itself a chunk-load fail, the same broken network bites the SDK fetch too.
3. **No UX fallback**. `ErrorFallback` gated all detail behind `isDev`. Users couldn't even note the error message to report it.

This PR closes (1) and provides a UX safety net for any case where capture still fails. Direct-beacon fallback for chunk-load failures (option (b) in the issue) is intentionally out of scope — wait-and-see after this lands.

## How

- `componentDidCatch` does `Promise.all([import("@sentry/react"), import("@/lib/sentry")])`, calls `initSentry()` (idempotent — guarded by DSN check, and Sentry's own `init` short-circuits on second call), then `captureException` with the `error_id` tag.
- `makeErrorId` seeds FNV-1a from `error.message` only. Including the stack initially seemed reasonable, but React's DEV-mode re-renders mint fresh `Error` instances with different stack frames each attempt — that broke ID stability across renders. Message-only signature also gives natural dedup ("err_abc123 again? same crash") and Sentry has the full event for triage anyway.
- `ErrorFallback` accepts new optional props (`errorId`, `componentStack`) plumbed from the boundary; falls back to `makeErrorId(error)` when called standalone (e.g. inline variant from elsewhere).
- Clipboard write uses `navigator.clipboard.writeText` with a legacy `execCommand` fallback for hostile environments. Sonner toasts on success/failure.

### Tests

- `src/lib/errorReport.test.ts` — 11 cases (id stability, format, message/no-stack edge cases, full report builder, markdown formatter)
- `src/components/AppErrorBoundary.test.tsx` — 4 cases (renders children, fallback ctx shape, `initSentry` called before `captureException` with `error_id` tag, resetError flow)
- `src/components/ErrorFallback.test.tsx` — extended: error ID badge in both variants, message visible in prod, details toggle, copy-report happy path + clipboard-failure path

Full suite: 1478 ✅, typecheck ✅, build ✅, lint clean on touched files.

Closes #344

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/lib/errorReport.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    return typeof __APP_VERSION__ === "string" ? __APP_VERSION__ : "unknown"
+  } catch {
+    return "unknown"
+  }
+}
+
+function readRoute(): string {
+  if (typeof window === "undefined") return ""
+  const { pathname, search, hash } = window.location
+  return `${pathname}${search}${hash}`
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/lib/errorReport.ts]: Good catch — `readRoute()` now returns `window.location.pathname` only. Pasting the report into a GH issue can no longer leak OAuth `code`, Supabase implicit-flow `#access_token`, or magic-link tokens. Trade-off accepted: we lose a bit of debug context (e.g. `?tab=balance`) but a denylist would just rot.

Addressed in `2334fc5` (2334fc5138f111a813ec2f4ee7f68a902119fb30)

File: src/components/ErrorFallback.tsx [hunk 2]
```diff
@@ -1,38 +1,141 @@
...
   const { t } = useTranslation("error")
   const [showDetails, setShowDetails] = useState(false)
-  const isDev = import.meta.env.DEV
+
+  const resolvedId = errorId ?? makeErrorId(error)
+  const report = buildErrorReport({
+    id: resolvedId,
+    error,
+    componentStack: componentStack ?? null,
+  })
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/components/ErrorFallback.tsx:71]: Right — the boundary now mints `caughtAt: Date` in `getDerivedStateFromError` and threads it through the fallback context. `ErrorFallback` consumes it via `useMemo(() => caughtAt ?? new Date(), [caughtAt])` so the timestamp stays stable across re-renders (toggle details, etc.) and reflects the actual catch moment. New test: "pins caughtAt to the moment of the catch (stable across re-renders)".

Addressed in `2334fc5` (2334fc5138f111a813ec2f4ee7f68a902119fb30)

File: src/components/ErrorFallback.tsx [hunk 3]
```diff
@@ -1,38 +1,141 @@
...
+    const ta = document.createElement("textarea")
+    ta.value = text
+    ta.setAttribute("readonly", "")
+    ta.style.position = "fixed"
+    ta.style.opacity = "0"
+    document.body.appendChild(ta)
+    ta.select()
+    const ok = document.execCommand("copy")
+    document.body.removeChild(ta)
+    return ok
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/components/ErrorFallback.tsx]: Done — appended the textarea before the try block, then `select()`/`execCommand` run inside try with `ta.remove()` in finally. Even if `select()` or `execCommand` throw, no orphan node leaks into the DOM.

Addressed in `2334fc5` (2334fc5138f111a813ec2f4ee7f68a902119fb30)

File: src/components/ErrorFallback.test.tsx [hunk 4]
```diff
@@ -37,23 +71,68 @@ describe("ErrorFallback", () => {
...
+      Object.defineProperty(navigator, "clipboard", {
+        value: {
+          writeText: () => Promise.reject(new Error("denied")),
+        },
+        configurable: true,
+      })
+      Object.defineProperty(document, "execCommand", {
+        value: () => false,
+        configurable: true,
+      })
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/components/ErrorFallback.test.tsx:180]: Fixed — capture the original descriptors of `navigator.clipboard` and `document.execCommand` (via `Object.getOwnPropertyDescriptor` on the prototypes) and restore them in `afterEach`. Stubs no longer leak across test files in the same Vitest worker.

Addressed in `2334fc5` (2334fc5138f111a813ec2f4ee7f68a902119fb30)

File: src/components/AppErrorBoundary.tsx [hunk 5]
```diff
@@ -1,54 +1,92 @@
...
+ * we ship zero Sentry bytes here.
+ *
+ * Two reliability fixes vs the original implementation:
+ *
+ * 1. We import `@/lib/sentry` alongside `@sentry/react` and call
+ *    `initSentry()` before `captureException`. `initSentry` is idempotent
+ *    (no-op when no DSN, and Sentry's own `init` short-circuits on second
+ *    call), but this closes the race where an error fires during the
+ *    `requestIdleCallback` window in `main.tsx` — without this guard the
+ *    capture is silently dropped because the global hub has no client.
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/components/AppErrorBoundary.tsx:36]: Real bug, fixed at the source. `src/lib/sentry.ts` now has a module-level `let initialized = false` flag that short-circuits subsequent calls. The boundary contract ("call it before captureException, idempotent") now actually holds — no more silent client replacement when both `main.tsx` (idle) and the error path call it. New test file `src/lib/sentry.test.ts` asserts `Sentry.init` is called exactly once across multiple `initSentry()` calls.

Addressed in `2334fc5` (2334fc5138f111a813ec2f4ee7f68a902119fb30)
